en/tutorials-and-examples/cms/authentication-plugin.rst

```diff
@@ -267,6 +267,37 @@ Add the logout action to the UsersController class::
 Now you can visit ``/users/logout`` to log out. You should then be sent to the login
 page.
 
+
+Customizing Login / Logout
+=========================
+
+You can also customize your login and logout screen very easily in CakePHP. 
+Add 'logout' as in the example below::
+
+    // in src/Controller/AppController.php
+    public function beforeFilter(\Cake\Event\EventInterface $event)
+    {
+        parent::beforeFilter($event);
+        // for all controllers in our application, make index and view
+        // actions public, skipping the authentication check
+        $this->Authentication->addUnauthenticatedActions(['index', 'view', 'logout']);
```

Comment:
Any reason you are going with this approach over allowing logout in the UsersController?

Reply:
I wanted a logout screen, so I did it (below) because I don't know how to do it differently. But you can teach me how to do it right.

   public function beforeFilter(\Cake\Event\EventInterface $event)
    {
        parent::beforeFilter($event);
        // configure the login action to don't require authentication, preventing
        // the infinite redirect loop issue
        $this->Authentication->addUnauthenticatedActions(['login']);
        $this->Authentication->addUnauthenticatedActions(['logout']);
    }

Reply:
How do I cancel this change?

    public function login() {
    public function login()

Reply:
you can remove on logout if you wish

Reply:
you have to use latest version to work on it
or now you can re-base it , or add it again with new commit

Reply:
You can add logout and login to the same addUnauthenticatedActions() call in the UsersController beforeFilter method.